query-engine/core/src/query_document/argument_value.rs

```diff
@@ -0,0 +1,67 @@
+use bigdecimal::BigDecimal;
+use indexmap::IndexMap;
+use prisma_models::PrismaValue;
+
+pub type ArgumentValueObject = IndexMap<String, ArgumentValue>;
+
+/// Represents the input values in a Document.
+/// This abstraction is mostly there to hold special kind of values such as `FieldRef` which have to be disambiguated at query-validation time.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ArgumentValue {
+    Scalar(PrismaValue),
+    Object(ArgumentValueObject),
+    List(Vec<ArgumentValue>),
+}
+
+impl ArgumentValue {
+    pub fn null() -> Self {
```

Comment:
I'm wondering, does this need to be `pub`? (I'm trying to figure out what the public API of each crate in QE actually is)

Reply:
I tried making them `pub(crate)`. It's not possible because it's used in `request-handlers` which isn't part of the `core` crate. (Unless I misunderstood something about pub(crate) visibility)

Reply:
There are a couple of them that _could_ be `pub(crate)`. But I'm wondering if it makes sense to only have a couple of constructors with reduced visibility

Reply:
No no, no worries. request_handlers is the piece of the puzzle I was missing. Just trying to understand what depends on what. I think we can close this thread, thanks!